onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc

```diff
@@ -182,6 +188,61 @@ bool GruOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, c
   return true;
 }
 
+bool GruOpBuilder::HasSupportedInputsImpl(const Node& node, const WebnnDeviceType /* device_type */,
+                                          const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& op_type = node.OpType();
+  int32_t input0_type;  // input data type
+  int32_t input1_type;  // weight data type
+  int32_t input2_type;  // recurrentWeight data type
+  int32_t input3_type;  // bias data type
+  int32_t input4_type;  // recurrentBias data type
+  int32_t input5_type;  // initialHiddenState data type
+  bool has_input3 = input_defs.size() > 3 && input_defs[3]->Exists();
+  bool has_input4 = input_defs.size() > 4 && input_defs[4]->Exists();
+  bool has_input5 = input_defs.size() > 5 && input_defs[5]->Exists();
+
+  if (!GetType(*input_defs[0], input0_type, logger) ||
+      !GetType(*input_defs[1], input1_type, logger) ||
+      !GetType(*input_defs[2], input2_type, logger) ||
+      (has_input3 && !GetType(*input_defs[3], input3_type, logger)) ||
+      (has_input4 && !GetType(*input_defs[4], input4_type, logger)) ||
+      (has_input5 && !GetType(*input_defs[5], input5_type, logger))) {
+    return false;
+  }
+
+  std::unordered_set<ONNX_NAMESPACE::TensorProto_DataType> supported_data_types;
+  if (device_type == WebnnDeviceType::CPU) {
```

Comment:
Are `opSupportLimits` mature enough now to use them, rather than hard-coding data types into the WebNN EP?

Reply:
I think there are still some basic codes that are not ready yet. Will use `opSupportLimits` later.

Reply:
Ok.

Reply:
Ok. Well we can update all the ops together once `opSupportLimits` are more mature in Chromium.